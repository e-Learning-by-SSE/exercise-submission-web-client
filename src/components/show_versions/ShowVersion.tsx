import React from "react";

export default class ShowVersion extends React.Component<React.PropsWithChildren<{}>> {
    
        constructor(props: React.PropsWithChildren<{}>) {
            super(props);
    
        }

    
        render(): React.ReactNode {
            return (
                <div className="show-version">
                    Version1 <p/>
                    Version2 <p/>
                    Version3 <p/>
                    Version4 <p/>
                    Version5 <p/>
                    Version6 <p/>
                    Version7 <p/>
                    Version8 <p/>
                    Version9 <p/>
                    Version10 <p/>
                </div>
            );
        }
    }