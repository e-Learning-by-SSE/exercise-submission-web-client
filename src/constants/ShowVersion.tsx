export const ShowVersionState = {
    LOADING: "Loading",
    SELECTASSIGNMENT: "SelectAssignment",
    DRAWTABLE: "DrawTable"
}