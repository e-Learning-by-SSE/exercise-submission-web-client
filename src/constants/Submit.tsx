export const SubmitState = {
    DROPFILE: "Dropfile",
    FILETREE: "Filetree",
    SELECTASSIGNMENT: "SelectAssignment",
    UPLOAD: "Upload",
    RESULT: "Result"
}