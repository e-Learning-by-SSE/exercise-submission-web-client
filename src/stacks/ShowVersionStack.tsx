import { StackInterface } from "../interface/StackInterface";

export const ShowVersionStack: StackInterface[] = [
{
    step: " View Versions",
    description: "View | Download Versions",
    id: "DrawTable",
    icon: "table",
    order: 1
},
{
    step: "Select Assignment",
    description: "Select assignment for Upload",
    id: "SelectAssignment",
    icon: "window maximize",
    order: 2
    }];
   