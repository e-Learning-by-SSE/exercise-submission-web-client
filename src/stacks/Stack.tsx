import React from "react";
import { StackInterface } from "../interface/StackInterface";
import {Step, Divider} from "semantic-ui-react";


export default class Stack extends React.Component<React.PropsWithChildren<{stack: StackInterface[], selected: string}>> {


    private createStepChain() {
        return(

            <Step.Group size="mini" width={this.props.stack.length} atached="top" fluid>
                {this.createSteps()}
            </Step.Group>

        );

    }

    private createSteps() {

        const steps = [];
        let stack = this.props.stack;
       

        let selectedStack = stack.find((stack) => stack.id === this.props.selected);
        if(selectedStack !== undefined) {
            let selectedOrder = selectedStack.order;
            for(let i = 0; i < stack.length; i++) {
             let step = stack.find((stack) => stack.order === i+1);

             if(step !== undefined) {
                let active = false;
                let completed = false;
                    if(step.id === selectedStack.id) {
                        active = true;
                    }
                    if(step.order < selectedOrder) {
                        completed = true;
                    }
                
                    steps.push(  
                    <Step active={active} completed={completed} key={i}>       
                    <Step.Content>
                        <Step.Title>{step.step}</Step.Title>       
                    </Step.Content>
                </Step>)
             }
             
         }
           
        }

        return steps;

    }
/* <Step active={active} completed={completed} key={i}>
                        <Icon name={step.icon} />
                    <Step.Content>
                        <Step.Title>{step.step}</Step.Title>
                        <Step.Description>{step.description}</Step.Description>
                    </Step.Content>*/


    render() {
        return (
            <div className="sequence-step">
                {this.createStepChain()}
                <Divider/>
            </div>
        )
    }

}