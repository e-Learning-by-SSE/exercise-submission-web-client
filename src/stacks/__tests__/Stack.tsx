import renderer from "react-test-renderer";
import { ShowVersionState } from "../../constants/ShowVersion";
import { ShowVersionStack } from "../ShowVersionStack";
import Stack from "../Stack";

it("render Stack",()   => {
    const stack = renderer.create(<Stack stack={ShowVersionStack} selected={ShowVersionState.DRAWTABLE}/>).toJSON();
    expect(stack).toMatchSnapshot();
});