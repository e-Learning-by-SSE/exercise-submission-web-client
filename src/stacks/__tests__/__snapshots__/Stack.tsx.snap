// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Stack 1`] = `
<div
  className="sequence-step"
>
  <div
    className="ui divider"
  />
  <div
    atached="top"
    className="ui mini fluid steps"
    width={2}
  >
    <div
      className="completed step"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="window maximize icon"
        onClick={[Function]}
      />
      <div
        className="content"
      >
        <div
          className="title"
        >
          Select Assignment
        </div>
        <div
          className="description"
        >
          Select assignment for Upload
        </div>
      </div>
    </div>
    <div
      className="active step"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="table icon"
        onClick={[Function]}
      />
      <div
        className="content"
      >
        <div
          className="title"
        >
           View Versions
        </div>
        <div
          className="description"
        >
          View | Download Versions
        </div>
      </div>
    </div>
  </div>
</div>
`;
